selfservice/hook/session_destroyer.go

```diff
@@ -0,0 +1,36 @@
+package hook
+
+import (
+	"net/http"
+
+	"github.com/ory/kratos/selfservice/flow/login"
+	"github.com/ory/kratos/selfservice/flow/registration"
+	"github.com/ory/kratos/session"
+)
+
+var _ login.PostHookExecutor = new(SessionDestroyer)
+
+type sessionDestroyerDependencies interface {
+	session.ManagementProvider
+	session.PersistenceProvider
+}
+
+type SessionDestroyer struct {
+	r sessionDestroyerDependencies
+}
+
+func NewSessionDestroyer(r sessionDestroyerDependencies) *SessionDestroyer {
+	return &SessionDestroyer{r: r}
+}
+
+func (e *SessionDestroyer) ExecuteRegistrationPostHook(w http.ResponseWriter, r *http.Request, a *registration.Request, s *session.Session) error {
+
+	return nil
+}
+
+func (e *SessionDestroyer) ExecuteLoginPostHook(w http.ResponseWriter, r *http.Request, a *login.Request, s *session.Session) error {
+	if err := e.r.SessionPersister().DeleteSession(r.Context(), s.Identity.ID); err != nil {
```

Comment:
`DeleteSession` accepts the session ID, so you can not use `s.Identity.ID` here. Instead, you need to add a new method to [`session.Persister`](https://github.com/ory/kratos/blob/master/session/persistence.go#L31) such as `DeleteSessionsFor(ctx context.Context, identityID uuid.UUID) error`.

Next, you need to implement a test for that function [around here](https://github.com/ory/kratos/blob/master/session/persistence.go#L67).

Next, you need to implement the actual storage logic [here](https://github.com/ory/kratos/blob/master/persistence/sql/persister_session.go#L22).

Once that is done, you should run:

```
go test -short -tags sqlite ./...
```

in the root directory of the project and see if tests pass.

Then, you will use that new method you implemented here.

Reply:
done